openedx/core/djangoapps/content_libraries/api/blocks.py

```diff
@@ -490,6 +512,115 @@ def import_staged_content_from_user_clipboard(library_key: LibraryLocatorV2, use
+    import it into the library, along with all its child XBlocks.
+    """
+    olx_node = etree.fromstring(olx_str)
+    if olx_node.tag != "vertical":
+        raise ValueError("This method is only designed to work with <vertical> XBlocks (units).")
+    # The olx_str looks like this:
+    # <vertical><block1>...[XML]...</block1><block2>...[XML]...</block2>...</vertical>
+    # Ideally we could split it up and preserve the strings, but that is difficult to do correctly, so we'll split
+    # it up using the XML nodes. This will unfortunately remove any custom comments or formatting in the XML, but that's
+    # OK since Studio-edited blocks won't have that anyways (hand-edited and library blocks can and do).
+
+    title = olx_node.attrib.get("display_name")
+    if not title:
+        # Not sure if we need this, but find a localized default title if none was set:
+        from cms.djangoapps.contentstore import helpers as studio_helpers
+        title = studio_helpers.xblock_type_display_name(olx_node.tag)
+
+    # Start an atomic section so the whole paste succeeds or fails together:
+    with transaction.atomic():
+        container = create_container(
+            library_key=library_key,
+            container_type=ContainerType.Unit,
+            slug=None,  # auto-generate slug from title
+            title=title,
+            user_id=user.id,
+        )
+        new_child_keys: list[UsageKeyV2] = []
+        for child_node in olx_node:
+            try:
+                child_metadata = _import_staged_block(
+                    block_type=child_node.tag,
+                    olx_str=etree.tostring(child_node, encoding='unicode'),
+                    library_key=library_key,
+                    source_context_key=source_context_key,
+                    user=user,
+                    staged_content_id=staged_content_id,
+                    staged_content_files=staged_content_files,
+                    now=now,
+                )
+                new_child_keys.append(child_metadata.usage_key)
+            except IncompatibleTypesError:
+                continue  # Skip blocks that won't work in libraries
+        update_container_children(container.container_key, new_child_keys, user_id=user.id)
+        # Re-fetch the container because the 'last_draft_created' will have changed when we added children
+        container = get_container(container.container_key)
+    return container
+
+
+def import_staged_content_from_user_clipboard(library_key: LibraryLocatorV2, user) -> PublishableItem:
+    """
+    Create a new library item from the staged content from clipboard.
+    Can create containers (e.g. units) or XBlocks.
+
+    Returns the newly created item metadata
+    """
+    from openedx.core.djangoapps.content_staging import api as content_staging_api
+    if not content_staging_api:
+        raise RuntimeError("The required content_staging app is not installed")
```

Comment:
Is it even possible to not install the `content_staging` app at this point? Is the concern that this could be called from the LMS instead of Studio?